docs/ADL2/master04.5-cadl_primitive_types.adoc

```diff
@@ -516,6 +619,41 @@ The _recommendation_ is that if a terminology constraint is not `required`, a pl
 
 Soft terminology constraints are specified in ADL using keywords prior to the formal constraint. Taking into account the recommendation for coded-text + text, the typical usage is as shown below. If no keyword is supplied, the meaning is `required`, although this may also be stated using the `required` keyword if desired.
 
+[tabs,sync-group-id=adl-example]
+====
+at-coded ADL2::
++
+[source, cadl]
+--------
+    items matches {
+        ELEMENT[at0010] occurrences matches {0..1} matches {
+            name matches {
+                DV_CODED_TEXT[at0007] matches {
+                    defining_code matches {preferred [at0004]}	-- prefer 'Substance', any code ok
+                }
+                DV_TEXT[at0008]                                -- or plain text
+            }
+            value matches {
+                DV_CODED_TEXT[at0054] matches {
+                    defining_code matches {example [ac1]}	-- ac1 provided as example only
+                }
+                DV_TEXT[at0055]                               -- or plain text
+            }
+        }
+        ELEMENT[at0021] occurrences matches {0..1} matches {
+            value matches {
+                DV_CODED_TEXT[at0057] matches {
+                    defining_code matches {extensible [ac2]} -- use ac2 value-set if there is a match
```

Comment:
Why _ac_ code in _at_ example?

Reply:
https://github.com/openEHR/specifications-AM/pull/16#discussion_r1670057543